### ISSUE TITLE
Gracefully skip loading when Vim version is below a minimum

### ISSUE THREAD

**Issue by ches** [ensime/ensime-vim#371]
See #370.

We're using a few parts of the Vim Python API that were added in Vim 7.4 (e.g. `vim.vars`). We either need to find these and change them to use methods that are compatible with older Vim (e.g. #370), or declare supported compatibility and skip loading on older versions.

I'm in favor of the latter, ensime-vim is the type of plugin people use for concentrated Scala work in an environment setup optimized for day-to-day development. Most such users will run a fairly recent (Neo)vim, and IMO this is in keeping with ENSIME's overall tinkerer philosophy. Installs of 7.3 are becoming increasingly obsolete in OS packages, and we have better things to do than wrestle with compatibility for a minority.

FWIW UltiSnips is precedent of a high-value Python-based plugin that supports only 7.4+.

The work to do here is, I think:

- Add a simple `if version < 704` check to `finish` with a friendly message to the user.
- Document the requirement where appropriate.

**Comment by ktonga** [comment]
I totally agree whit everything you just said, adding modern-IDEs-like features to Vim is hard enough already without supporting every single version out there. So I should not focus on this kind of compatibility issues. But I would be keen to consider merging a PR if someone interested in such support come up with any. Obviously if it doesn't complicates the codebase too much.
